### PR TITLE
Modularize code to support n imus.

### PR DESCRIPTION
**Pull request by peasant98**
This PR modularizes so the user can specify any amount of imus they want on the robot. Furthermore, the data collection code takes care of n imus instead of the original 7 hardcoded ones.

Also, the imu optimization visualization now occurs on all imus at once:

All IMUs

### CODE REVIEW

File: scripts/data_generators/ConstantRotationDataSaver.py [hunk 1]
```diff
@@ -191,13 +191,15 @@ def __init__(self, controller, poses_list, filepath='data/constant_data'):
...
         # constant
-        # TODO: get imu names automatically
         self.pose_names = [pose[2] for pose in poses_list]
         self.joint_names = self.controller.joint_names
-        self.imu_names = ['imu_link0', 'imu_link1', 'imu_link2',
-                          'imu_link3', 'imu_link4', 'imu_link5', 'imu_link6']
-        self.imu_topics = ['imu_data0', 'imu_data1', 'imu_data2',
-                           'imu_data3', 'imu_data4', 'imu_data5', 'imu_data6']
+        all_topics = rospy.get_published_topics()
+        total_imu_topics = len([topic for topic in all_topics if 'imu' in topic[0]])
```
Comment (watakandai) [inline scripts/data_generators/ConstantRotationDataSaver.py:197]: smart, but what if we have other topics which have `imu` included in the topic names

File: scripts/imu_spawners/spawn_estimated_imu.py [hunk 2]
```diff
@@ -192,7 +192,7 @@ def load_estimated_poses(filename):
 if __name__ == '__main__':
     # rospy.init_node("set_estimated_imu_positions")
     n_joint = 7
-    n_imu = 7
+    n_imu = 6
```
Comment (watakandai) [inline scripts/imu_spawners/spawn_estimated_imu.py:195]: Are we able to modify these two lines to read from the save data file?

File: scripts/imu_spawners/spawn_estimated_imu.py [hunk 3]
```diff
@@ -238,14 +238,33 @@ def load_estimated_poses(filename):
     poses = np.zeros((7, 7))
```
Comment (watakandai) [inline scripts/imu_spawners/spawn_estimated_imu.py:238]: Same here. Can we automatically read from `Controller`?

File: scripts/imu_spawners/spawn_estimated_imu.py (hunk 2 above)
Comment (peasant98) [reply scripts/imu_spawners/spawn_estimated_imu.py:195]: yes, I will do that.

File: scripts/imu_spawners/spawn_estimated_imu.py (hunk 3 above)
Comment (peasant98) [reply scripts/imu_spawners/spawn_estimated_imu.py:239]: yes

Comment (peasant98) [reply scripts/imu_spawners/spawn_estimated_imu.py]: done

File: scripts/data_generators/ConstantRotationDataSaver.py (hunk 1 above)
Comment (peasant98) [reply scripts/data_generators/ConstantRotationDataSaver.py:197]: yea, would it be better to read from `imu_poses.txt`, which really indicates the amount of topics?

Comment (watakandai) [reply scripts/data_generators/ConstantRotationDataSaver.py:197]: yeah i gree.

**Pull request merged**